packages/govuk-frontend/src/govuk/common/configuration.mjs

```diff
@@ -0,0 +1,334 @@
+import { ConfigError } from '../errors/index.mjs'
+import { GOVUKFrontendComponent } from '../govuk-frontend-component.mjs'
+
+import { isObject, formatErrorMessage } from './index.mjs'
+
+/**
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @virtual
+ * @template {ObjectNested} [ConfigurationType={}]
+ * @template {Element} [RootElementType=HTMLElement]
+ * @augments GOVUKFrontendComponent<RootElementType>
+ */
+export class GOVUKFrontendComponentConfigurable extends GOVUKFrontendComponent {
```

Comment:
**nitpick** Something we can sort outside of the refactoring PRs is the naming of the class and whether we want to frontload with `Component` as it is now, or make it read like English with either `GOVUKFrontendConfigurableComponent` or `ConfigurableGOVUKFrontendComponent` 😊